pkg/planner/core/rule_decorrelate.go

```diff
@@ -160,7 +160,15 @@ func pruneRedundantApply(p base.LogicalPlan) (base.LogicalPlan, bool) {
 	if apply.JoinType != logicalop.LeftOuterJoin && apply.JoinType != logicalop.LeftOuterSemiJoin {
 		return nil, false
 	}
-
+	// add a strong limit for fix the https://github.com/pingcap/tidb/issues/58451. we can remove it when to have better implememnt.
+	// But this problem has affected tiflash CI.
+	applyChildren := apply.Children()
+	if len(applyChildren) >= 2 {
+		corCols := coreusage.ExtractCorColumnsBySchema4LogicalPlan(applyChildren[1], applyChildren[0].Schema())
```

Comment:
Can you explain the root cause and why this fix works? More details are appreciated